### PR TITLE
Vinaybackend

### PR DESCRIPTION
**Pull request by VINERAJ**
uh I forgot to create pull requests for the past couple days. I added riot api calls in case we need them, and added a method to delete the user

### CODE REVIEW

File: Project/backend/java-server/app/src/main/java/org/server/DBController.java [hunk 1]
```diff
@@ -114,6 +115,27 @@ public void addUser(User user) {
...
     }
 
+    public boolean deleteUser(User user) {
+        Connection connection = null;
+        try {
+            connection = DriverManager.getConnection(dbPath);
+            Statement stmt = connection.createStatement();
+            String sq1 = "DELETE FROM users WHERE uuid = '" + user + "'";
+            stmt.executeQuery(sq1);
+            JOptionPane.showMessageDialog(null, "Deleted User");
```
Comment (adhi-thirumala) [inline Project/backend/java-server/app/src/main/java/org/server/DBController.java:125]: why did we import a gui library?????

File: Project/backend/java-server/app/src/main/java/org/server/UserController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+public class UserController {
+    private static final String DB_URL = "jdbc:sqlite:main.db";
+
+    DBController db = new DBController(DB_URL);
+    @PostMapping("/api/v1/_delete_user_")
+    public void deleteUser(@PathVariable String uuid) {
+        User user = db.getUser(uuid, true);
+        boolean deleted = db.deleteUser(user);
+        if (deleted) {
+            System.out.println("Deleted user: " + uuid);
```
Comment (adhi-thirumala) [inline Project/backend/java-server/app/src/main/java/org/server/UserController.java:22]: should be a return value + status code

File: Project/backend/java-server/app/src/main/java/org/server/UserController.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        }
+    }
+    @PostMapping("/api/v1/add_user")
+    public void addUser(@PathVariable String uuid, @PathVariable String firstName, @PathVariable String lastName, @PathVariable String riotID, @PathVariable String discordID, @PathVariable String[] oneWayMatched, @PathVariable String[] twoWayMatched) {
```
Comment (adhi-thirumala) [inline Project/backend/java-server/app/src/main/java/org/server/UserController.java:26]: these are all path variables, but no place for them in the route

**Review by adhi-thirumala — CHANGES_REQUESTED** [review]
fix changes in the comments + write unit tests for methods

**Comment by yangr0** [comment]
@VINERAJ 
1. Never force push, it breaks a lot of previous commits. If git is blocking a commit, it's probably for a good reason. Use "git status" to figure out the problem. 
2. Make sure you are pulling from the master branch every time there is a new commit pushed to the master branch to avoid merge conflicts. Do it by running: "git pull origin master". 
3. Do not create a branch like "Vinaybackend", name it something like "delete-user-route". Branches should be created to complete one specific objective only and then destroyed after it has finished it's objective. 

I made a tutorial about the entire process of making commits and creating branches to this repository. Check the group 31 channel on Discord for the video.

**Comment by adhi-thirumala** [comment]
still needs above changes

**Pull request closed without merging**